daemon/start.go

```diff
@@ -78,8 +78,10 @@ func (daemon *Daemon) ContainerStart(name string, hostConfig *containertypes.Hos
 	}
 	// Adapt for old containers in case we have updates in this function and
 	// old containers never have chance to call the new function in create stage.
-	if err := daemon.adaptContainerSettings(container.HostConfig, false); err != nil {
-		return err
+	if container.HostConfig != nil {
```

Comment:
This should be `if hostConfig != nil {` instead (we want to check the `hostConfig` passed as start here and not the current `container.HostConfig`).